india_compliance/gst_india/report/gstr_1_document_issued_summary/gstr_1_document_issued_summary.py

```diff
@@ -0,0 +1,196 @@
+        )
+
+        naming_series_data.setdefault(
+            naming_series, {"canceled_count": 0, "total_count": 0, "document_names": {}}
+        )
+
+        names = naming_series_data.get(naming_series)
+        if item.docstatus == 2:
+            names["canceled_count"] += 1
+
+        names["document_names"][item.name] = item.creation
+        names["total_count"] += 1
+
+    res = []
+    for naming_series, name_data in naming_series_data.items():
+        if not name_data:
+            continue
+
+        sorted_names = sorted(name_data["document_names"].items(), key=lambda x: x[1])
+        if sorted_names and len(sorted_names[0]) > 0:
+            res.append(
+                frappe._dict(
+                    {
+                        "naming_series": naming_series,
+                        "nature_of_document": nature_of_document,
+                        "from_serial_no": sorted_names[0][0],
+                        "to_serial_no": sorted_names[len(sorted_names) - 1][0],
+                        "total_number": name_data.get("total_count"),
+                        "canceled": name_data.get("canceled_count"),
+                        "total_issued": name_data.get("canceled_count")
+                        + name_data.get("total_count"),
+                    }
+                )
+            )
+
+    return res
+
+
+def apply_document_wise_condition(filters, query, nature_of_document, doctype):
+    if doctype == "Purchase Invoice" and filters.get("company_address"):
+        query = query.where(doctype.shipping_address == filters.company_address)
+    elif filters.get("company_gstin"):
+        query = query.where(doctype.company_gstin == filters.company_gstin)
+
+    bank_accounts = get_bank_accounts(filters.company)
+
+    if nature_of_document == "Invoices for outward supply":
+        query = query
+    elif nature_of_document == "Invoices for inward supply":
+        query = query.where(doctype.gst_category != "Unregistered")
+    elif nature_of_document == "Invoices for inward supply from unregistered person":
+        query = query.where(doctype.gst_category == "Unregistered")
+    elif nature_of_document == "Debit Note":
+        query = query.where(doctype.is_return == 1)
+    elif nature_of_document == "Credit Note":
+        query = query.where(doctype.is_return == 1)
+    elif nature_of_document == "Receipt Voucher":
+        query = query.where(doctype.payment_type == "Receive")
+    elif nature_of_document == "Payment Voucher":
+        query = query.where(doctype.payment_type == "Pay")
+    elif nature_of_document == "Receipt Voucher (JV)":
+        journal_entries = frappe.db.get_all(
+            "Journal Entry Account",
+            filters={"account": ("in", bank_accounts), "debit": (">", 0), "credit": 0},
+            pluck="parent",
+        )
+        query = query.where(doctype.name.isin(journal_entries)).where(
+            doctype.voucher_type != "Contra Entry"
+        )
```

Comment:
This may not be correct as all JV through banks are not required to be reported.
Only advance payments against Sales orders or Refunds of such advance payments are required.

Hence, best if we restrict this to Payment entry.
Also, this can be further reduced to instances where GST is paid from Payment Entry.